src/Acuminator/Acuminator.Analyzers/Analyzers/DAC/DacPropertyAttributes/DacExtensionDefaultAttributeAnalyzer.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Acuminator.Utilities;
+using System.Collections.Immutable;
+using PX.Data;
+using System.Threading;
+
+namespace Acuminator.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	class DacExtensionDefaultAttributeAnalyzer : PXDiagnosticAnalyzer
+	{
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create
+			(
+				Descriptors.PX1030_DefaultAttibuteToExisitingRecords
+			);
+#pragma warning disable CS4014
+		internal override void AnalyzeCompilation(CompilationStartAnalysisContext compilationStartContext, PXContext pxContext)
+		{
+			compilationStartContext.RegisterSymbolAction(symbolContext =>
+				AnalyzePropertyAsync(symbolContext, pxContext), SymbolKind.Property);
+		}
+#pragma warning restore CS4014
+		private static async Task AnalyzePropertyAsync(SymbolAnalysisContext symbolContext, PXContext pxContext)
+		{
+			IPropertySymbol property = symbolContext.Symbol as IPropertySymbol;
+			AttributeInformation attributeInformation = new AttributeInformation(pxContext);
+
+			if (!CheckProperty(property, pxContext))
+				return;
+
+			ImmutableArray<AttributeData> attributes = property.GetAttributes();
+
+			if (attributes.Length == 0 || symbolContext.CancellationToken.IsCancellationRequested)
+				return;
+
+			
+
+			if (symbolContext.CancellationToken.IsCancellationRequested)
+				return;
+			
+			bool isBoundField = attributeInformation.AreBoundAttributes(attributes.Select(a => a.AttributeClass).ToList());
+			if (isBoundField)
+			{
+				if (IsIBqlTableTypeImplementation(property, pxContext)) // BQLTable class bound field
+					return;
+				foreach (var attribute in attributes)
+				{
+					var typesHierarchy = attribute.AttributeClass.GetBaseTypesAndThis();
+					if(typesHierarchy.Contains(pxContext.AttributeTypes.PXDefaultAttribute))
+					{
+						foreach (var argument in attribute.NamedArguments)
+						{
+							if (argument.Key.Contains("PersistingCheck") && (int)argument.Value.Value == (int)PXPersistingCheck.Nothing)
+								return;
+						}
+						Location[] locations = await Task.WhenAll(GetAttributeLocationAsync(attribute, symbolContext.CancellationToken));
```

Comment:
@saloponov ConfigureAwait(false) must be used in all await expressions

Reply:
@SENya1990 @saloponov it is totally correct for the analyzers & code fixes, but you should be aware that in other places `ConfigureAwait(false)` is not the best choice. See [this article](https://github.com/Microsoft/vs-threading/blob/master/doc/cookbook_vs.md) for more details.